src/auth/ThreeStepOwned.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+/// @notice Three-step single owner authorization mixin.
+/// @author SolBase (https://github.com/Sol-DAO/solbase/blob/main/src/auth/ThreeStepOwned.sol)
+/// @author Modified from Solmate (https://github.com/transmissions11/solmate/blob/main/src/auth/Owned.sol)
+abstract contract ThreeStepOwned {
+    /*//////////////////////////////////////////////////////////////
+                                 EVENTS
+    //////////////////////////////////////////////////////////////*/
+
+    event OwnerUpdateInitiated(address indexed user, address indexed ownerCandidate);
+
+    event OwnerUpdated(address indexed user, address indexed newOwner);
+
+    /*//////////////////////////////////////////////////////////////
+                            OWNERSHIP STORAGE
+    //////////////////////////////////////////////////////////////*/
+
+    address public owner;
+    
+    address internal _ownerCandidate;
+    
+    bool internal _ownerCandidateConfirmed;
```

Comment:
hmm - are you guys sure that you want to use a `bool` type here? A `bool` is `uint8` under the hood and we could pack `owner`, `_ownerCandidate` and `_ownerCandidateConfirmed` into exactly a 32-byte slot by declaring it a `uint16` (160+160+16=256). The flags of `_ownerCandidateConfirmed` are `1` and `2` as in the reentrancyGuard to save gas. But honestly haven't checked myself - so just raising the point here.

Reply:
I thought that as long as the fields do not exceed the slot size, they will be packed into the same one. So I'd expect:

```solidity
    address public owner; // slot 0

    address internal _ownerCandidate; // slot 1

    bool internal _ownerCandidateConfirmed; // also slot 1
```

If that's not the case (but I think it is!), I'm completely in favor of switching the `bool` to a `uint96`, which will do the 2nd slot to be completely filled and using `0` and `1` instead as `false` and `true` for the flag (cheaper than `1` and `2` as zero is the default value).

But as I said, I think won't do any difference and it's already as packed as possible for this case.

Reply:
I actually made a mistake in my calculations lol (too much in my brain going on lol) - too sleepy. You're right here!